File: Problema 1/src2/gui.py
import tkinter as tk
import time  # Importação correta para o uso do time.time()

class TreinoNadadorGUI:
    def __init__(self, master, timer):
        self.master = master
        self.master.title("Registro de Tempos do Nadador")
        self.timer = timer
        
        self.label_instrucoes = tk.Label(master, text="Pressione 'Iniciar' para começar o treino.")
        self.label_instrucoes.pack(pady=10)

        self.botaoiniciar = tk.Button(master, text="Iniciar", command=self.iniciar_treino)
        self.botaoiniciar.pack(pady=5)

        self.botaoregistrar = tk.Button(master, text="Registrar Tempo", command=self.registrar_tempo, state=tk.DISABLED)
        self.botaoregistrar.pack(pady=5)

        self.label_tempo = tk.Label(master, text="Tempo: 0.00 segundos", font=("Helvetica", 14))
        self.label_tempo.pack(pady=20)

        self.botaofinalizar = tk.Button(master, text="Finalizar Treino", command=self.finalizar_treino, state=tk.DISABLED)
        self.botaofinalizar.pack(pady=5)

        self.label_resultados = tk.Label(master, text="")
        self.label_resultados.pack(pady=10)

    def atualizar_tempo(self):
        if self.timer.contador_ativo:
            tempo_atual = time.time() - self.timer.tempo_inicial
            self.label_tempo.config(text=f"Tempo: {tempo_atual:.2f} segundos")
            self.master.after(100, self.atualizar_tempo)  # Atualiza a cada 100ms

    def iniciar_treino(self):
        self.timer.iniciar()
        self.botaoiniciar.config(state=tk.DISABLED)
        self.botaoregistrar.config(state=tk.NORMAL)
        self.botaofinalizar.config(state=tk.NORMAL)
        self.label_resultados.config(text="")
        self.atualizar_tempo()  # Começa a atualizar o tempo exibido

    def registrar_tempo(self):
        tempo_registrado = self.timer.registrar_tempo()
        if tempo_registrado is not None:
            self.label_tempo.config(text=f"Tempo da volta: {tempo_registrado} segundos")
            if len(self.timer.tempos) == 10:
                self.botaoregistrar.config(state=tk.DISABLED)
                self.label_tempo.config(text="10 voltas registradas!")

    def finalizar_treino(self):
        tempos = self.timer.finalizar()
        if tempos:
            media, maior, menor = self.timer.calcular_estatisticas()
            resultados = (
                f"Tempos registrados: {tempos}\n"
                f"Média dos tempos: {media:.2f} segundos\n"
                f"Maior tempo: {maior:.2f} segundos\n"
                f"Menor tempo: {menor:.2f} segundos"
            )
            self.label_resultados.config(text=resultados)
            self.botaoiniciar.config(state=tk.NORMAL)
            self.botaofinalizar.config(state=tk.DISABLED)

File: Problema 1/JavaSolution/GUI.java
import javax.swing.*;
import java.awt.*;
import java.util.ArrayList;

public class GUI {
    private JFrame frame = new JFrame("Registro de Tempos do Nadador");
    private JLabel labelTempo = new JLabel("Tempo: 0.00 segundos");
    private JLabel labelResultados = new JLabel();
    private JButton botaoIniciar = new JButton("Iniciar");
    private JButton botaoRegistrar = new JButton("Registrar Tempo");
    private Timerer timerer;
    private Timer timer;
    private int voltas = 0;

    public GUI(Timerer timerer) {
        this.timerer = timerer;
        configurarJanela();
        configurarAcoes();
    }

    private void configurarJanela() {
        frame.setSize(850, 250);
        frame.setLayout(new GridLayout(5, 1, 10, 10));
        //5 linhas, 1 coluna, 10 pixels de espaçamento horizontal e 10 pixels de espaçamento vertical
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

        frame.add(new JLabel("Pressione 'Iniciar' para começar o treino."));
        frame.add(botaoIniciar);
        frame.add(botaoRegistrar);
        frame.add(labelTempo);
        frame.add(labelResultados);

        botaoRegistrar.setEnabled(false);
        frame.setVisible(true);
    }

    private void configurarAcoes() {
        botaoIniciar.addActionListener(e -> iniciarTreino());
        botaoRegistrar.addActionListener(e -> registrarTempo());
        //e -> é uma expressão lambda que define um ActionListener.
    }

    private void iniciarTreino() {
        timerer.iniciar();
        voltas = 0;
        botaoIniciar.setEnabled(false);
        botaoRegistrar.setEnabled(true);
        labelResultados.setText("");

        timer = new Timer(100, e -> atualizarTempo());
        timer.start();
    }

    private void atualizarTempo() {
        double tempoAtual = (System.currentTimeMillis() - timerer.tempoInicial) / 1000.0;
        labelTempo.setText(String.format("Tempo: %.2f segundos", tempoAtual));
    }

    private void registrarTempo() {
        Double tempo = timerer.registrarTempo();
        if (tempo != null) {
            labelTempo.setText(String.format("Tempo da volta: %.2f segundos", tempo));
            if (++voltas == 10) finalizarTreino();
        }
    }

    private void finalizarTreino() {
        labelTempo.setText("Treino finalizado!");
        botaoRegistrar.setEnabled(false);
        botaoIniciar.setEnabled(true);
        timer.stop();

        ArrayList<Double> tempos = timerer.finalizar();
        double[] stats = timerer.calcularEstatisticas();
        String resultados = String.format(
            "Tempos:  %s / Média: %.2f s Maior: %.2f s Menor: %.2f s", //o que é %s? r:
            tempos.toString().replaceAll("[\\[\\]]", ""), stats[0], stats[1], stats[2] //como posso fazer para que não exiba os colchetes?
        );
        labelResultados.setText(resultados);
    }
}
